scripts/check-toffee-plum.sh

```diff
@@ -21,39 +23,69 @@ function status_code() {
 
 function failure_check() {
     if [[ $statuscode != 200 ]] && [[ $1 == "Toffee" ]]; then
+        failure_msg_toffee="\n>:red_circle:  <$url| $ENV> is unhealthy"
+        # failure_msg_toffee+="\n>:green_circle:  <$url| $ENV> is healthy"
         failures_exist_toffee="true"
-        printf "\n>:red_circle:  <$url| $ENV> is unhealthy" >>slack-message.txt
+
+
     elif [[ $statuscode != 200 ]] && [[ $1 == "Plum" ]]; then
+        failure_msg_plum+="\n>:red_circle:  <$url| $ENV> is unhealthy"
         failures_exist_plum="true"
-        printf "\n>:red_circle:  <$url| $ENV> is unhealthy" >>slack-message.txt
     fi
 }
 
 function uptime() {
     for ENV in ${ENVIRONMENTS[@]}; do
         status_code $1
-        failure_check $1
+        failure_check $1 
     done
 }
 
-function do_failures_exist() {
-    if [[ $1 = "Toffee" ]]; then
-        if [[ $failures_exist_toffee != "true" ]]; then
-            printf "\n>:green_circle:  All environments in $1 are healthy" >>slack-message.txt
-        fi
-    elif [[ $1 = "Plum" ]]; then
-        if [[ $failures_exist_plum != "true" ]]; then
-            printf "\n>:green_circle:  All environments in $1 are healthy" >>slack-message.txt
-        fi
+function check_status() {
+    add_environments $1
+    uptime $1
+}
+
+function success_msg() {
+    printf "\n>:green_circle:  All environments are healthy" >>slack-message.txt
+}
+
+function format_failure() {
+    printf "\n*Toffee Status:*" >>slack-message.txt
+    if [[ $failures_exist_toffee == "true" ]]; then
+        printf '%s\n' "${failure_msg_toffee[@]}" >>slack-message.txt
+    else
+        success_msg
+    fi
+
+    printf "\n*Plum Status:*" >>slack-message.txt
+    if [[ $failures_exist_plum == "true" ]]; then
+        printf '%s\n' "${failure_msg_plum[@]}" >>slack-message.txt
+    else
+        success_msg
     fi
 }
 
-printf "\n:detective-pikachu: _*Check Toffee/Plum Status*_ \n" >>slack-message.txt
+function format_status() {
+    # if failure exists
+    if [[ $failures_exist_toffee == "true" ]] || [[ $failures_exist_plum == "true" ]]; then
```

Comment:
```suggestion
    if [[ $failures_exist_toffee || $failures_exist_plum ]]; then
```
Maybe? Bit easier to read quickly